features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewManagerScope.kt

```diff
@@ -114,7 +113,7 @@ internal class RumViewManagerScope(
         while (iterator.hasNext()) {
             val childScope = iterator.next()
             if (event is RumRawEvent.StopView) {
-                if (childScope.isActive() && (childScope as? RumViewScope)?.keyRef?.get() == event.key) {
+                if (childScope.isActive() && (childScope as? RumViewScope)?.key == event.key) {
```

Comment:
Should we compare `key.id` here instead? in theory 2 different APIs can create keys in a different manner: `RumScopeKey.from(key)` and `RumScopeKey.from(key, name)` and while they are backed by the same `key` (hence same `id`), the `name` component will be different, leading to `(childScope as? RumViewScope)?.key != event.key`. Such misuse will be hard to catch.